### PR TITLE
Fix cursor navigation on utf8

### PR DESCRIPTION
**Pull request by LeaNumworks**
Fix Left/Right/Backspace on special UTF8 glyphs ('é' for instance)

### CODE REVIEW

File: ion/src/shared/unicode/utf8_decoder.cpp [hunk 1]
```diff
@@ -63,6 +63,33 @@ CodePoint UTF8Decoder::previousCodePoint() {
...
+    followingCodePoint = nextCodePoint();
+  }
+  m_stringPosition = resultGlyphPosition;
+  return resultGlyphPosition;
+}
+
+const char * UTF8Decoder::previousGlyphPosition() {
+  assert(m_stringPosition > m_string);
+  CodePoint previousCP = previousCodePoint();
+  if (m_stringPosition == m_string) {
```
Comment (EmilieNumworks) [inline ion/src/shared/unicode/utf8_decoder.cpp]: Why do you need this special case? Won't it work the same without?

File: escher/src/text_input.cpp [hunk 2]
```diff
@@ -99,3 +99,22 @@ bool TextInput::removeEndOfLine() {
 bool TextInput::privateRemoveEndOfLine() {
   return contentView()->removeEndOfLine();
 }
+
+bool TextInput::moveCursorLeftRight(bool right) {
```
Comment (EmilieNumworks) [inline escher/src/text_input.cpp]: Why would you factorize moveCursorLeft and moveCursorRight if their implementation is a total disjunction?

File: ion/src/shared/unicode/utf8_decoder.cpp (hunk 1 above)
Comment (LeaNumworks) [reply ion/src/shared/unicode/utf8_decoder.cpp]: That's right :)

File: escher/src/text_input.cpp (hunk 2 above)
Comment (LeaNumworks) [reply escher/src/text_input.cpp]: That's right too ! It was not a disjunction when I started cleaning.

**Pull request merged**